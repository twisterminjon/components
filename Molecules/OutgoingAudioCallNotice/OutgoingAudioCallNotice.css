.outgoingaudiocallnotice-wrap {
  position: fixed;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: var(--z-callnotice);
}

.outgoingaudiocallnotice-dimmer {
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 21;

  background-color: var(--call_incoming_color);
}

.outgoingaudiocallnotice-content {
  position: absolute;
  z-index: 22;
}

.outgoingaudiocallnotice-buttons {
  position: relative;
  width: 300px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.outgoingaudiocallnotice-buttons-center {
  justify-content: space-around;
  /* Note sure why the margin is required but it won't center without it */
  margin: 0 auto;
}

.outgoingaudiocallnotice-buttons-spaced {
  justify-content: space-between;
  /* Note sure why the margin is required but it won't center without it */
  margin: 0 auto;
}

.outgoingaudiocallnotice-placeholder {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  width: 300px;
  height: 290px;
}

.outgoingaudiocallnotice-placeholder p {
  font-size: var(--font-base-body--size);
  color: var(--brandcolor);
}

.outgoingaudiocallnotice-calldetails {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  width: 400px;
  height: 400px;
}

.outgoingaudiocallnotice-calldetails p {
  color: var(--white);
  text-align: center;
}
.outgoingaudiocallnotice-calldetails-text {
  font-size: var(--font-base-body--size-medium);
}

.outgoingaudiocallnotice-calldetails-main {
  font-size: var(--font-base-body--size-xlarge) !important;
}
