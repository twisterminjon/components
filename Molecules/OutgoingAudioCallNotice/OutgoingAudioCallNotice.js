import React, { Component } from 'react'
import PropTypes from 'prop-types'

import UserCardLarge from '../../Molecules/UserCardLarge/UserCardLarge'

import './OutgoingAudioCallNotice.css'

export default class OutgoingAudioCallNotice extends Component {
  static propTypes = {
    /** Then name of the person being called */
    callingName: PropTypes.string,
    /** Title of the person calling  */
    title: PropTypes.string,
    /** the url to the callers profile pic */
    profileImage: PropTypes.string,
    /** If true, will display the form */
    show: PropTypes.bool.isRequired,
  }
  static defaultProps = {
    seekingNewCallee: false,
    title: '',
    callingName: '',
    profileImage: '',
    onSkip: () => {},
    showSkipCall: false,
  }

  render() {
    const { callingName, show, title, profileImage } = this.props

    if (!show) return null
    return (
      <div className="outgoingaudiocallnotice-wrap">
        <div className="outgoingaudiocallnotice-dimmer" />
        <div className="outgoingaudiocallnotice-content" data-testid="outgoing-audio-call-notice">
          <div className="outgoingaudiocallnotice-calldetails">
            <div style={{ height: 300, marginBottom: 10 }}>
              <UserCardLarge
                userName={callingName}
                title={title}
                profileImage={profileImage}
                actionText={'Calling...'}
              />
            </div>
            <div className="outgoingaudiocallnotice-placeholder">
              <p className="outgoingaudiocallnotice-calldetails-main">Your audio call is in progress.</p>
              <p className="outgoingaudiocallnotice-calldetails-text">
                When connected, you can join the call on your phone.
              </p>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
