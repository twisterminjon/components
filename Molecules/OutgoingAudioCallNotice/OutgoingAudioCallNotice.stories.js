import React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'

import OutgoingAudioCallNotice from './OutgoingAudioCallNotice'

const stories = storiesOf('Provider/v1/Molecules/CallNotices', module)
stories.addDecorator(withKnobs)

stories.add('OutgoingAudioCallNotice', () => {
  const callingNameVal = text('callingName', 'Becky Burrito')
  const showVal = boolean('show', true)

  const picVal = boolean('Show picture', false)
  const picSource = picVal ? 'https://tinyfac.es/data/avatars/852EC6E1-347C-4187-9D42-DF264CCF17BF-200w.jpeg' : ''

  return (
    <div
      style={{
        backgroundImage: 'url("https://picsum.photos/500/500/?random")',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
      }}>
      <OutgoingAudioCallNotice callingName={callingNameVal} profileImage={picSource} show={showVal} />
    </div>
  )
})
