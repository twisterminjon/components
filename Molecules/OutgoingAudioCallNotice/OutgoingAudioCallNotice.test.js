import React from 'react'
import OutgoingAudioCallNotice from './OutgoingAudioCallNotice'
import fakeImage from '../../../Images/ent-placeholder.png'

describe('OutgoingAudioCallNotice', () => {
  const name = 'Korben'
  const show = true

  it('matches the snapshot with skip call showing', () => {
    const wrapper = window.shallow(
      <OutgoingAudioCallNotice callingName={name} title="title" profilePic={fakeImage} show={show} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
