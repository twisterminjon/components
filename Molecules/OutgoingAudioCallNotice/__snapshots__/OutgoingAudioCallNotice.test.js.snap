// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutgoingAudioCallNotice matches the snapshot with skip call showing 1`] = `
<div
  className="outgoingaudiocallnotice-wrap"
>
  <div
    className="outgoingaudiocallnotice-dimmer"
  />
  <div
    className="outgoingaudiocallnotice-content"
    data-testid="outgoing-audio-call-notice"
  >
    <div
      className="outgoingaudiocallnotice-calldetails"
    >
      <div
        style={
          Object {
            "height": 300,
            "marginBottom": 10,
          }
        }
      >
        <UserCardLarge
          actionText="Calling..."
          onClick={[Function]}
          profileImage=""
          title="title"
          userName="Korben"
        />
      </div>
      <div
        className="outgoingaudiocallnotice-placeholder"
      >
        <p
          className="outgoingaudiocallnotice-calldetails-main"
        >
          Your audio call is in progress.
        </p>
        <p
          className="outgoingaudiocallnotice-calldetails-text"
        >
          When connected, you can join the call on your phone.
        </p>
      </div>
    </div>
  </div>
</div>
`;
